demos/features/niimath.command.html

```diff
@@ -1,120 +1,267 @@
+ -atan  : arc tangent function
+ -sqr   : square
+ -sqrt  : square root
+ -recip : reciprocal (1/current image)
+ -abs   : absolute value
+ -bin   : use (current image>0) to binarise
+ -binv  : binarise and invert (binarisation and logical inversion)
+ -fillh : fill holes in a binary mask (holes are internal - i.e. do not touch the edge of the FOV)
+ -fillh26 : fill holes using 26 connectivity
+ -index : replace each nonzero voxel with a unique (subject to wrapping) index number
+ -grid <value> <spacing> : add a 3D grid of intensity <value> with grid spacing <spacing>
+ -edge  : edge strength
+ -tfce <H> <E> <connectivity>: enhance with TFCE, e.g. -tfce 2 0.5 6 (maybe change 6 to 26 for skeletons)
+ -tfceS <H> <E> <connectivity> <X> <Y> <Z> <tfce_thresh>: show support area for voxel (X,Y,Z)
+ -nan   : replace NaNs (improper numbers) with 0
+ -nanm  : make NaN (improper number) mask with 1 for NaN voxels, 0 otherwise
+ -rand  : add uniform noise (range 0:1)
+ -randn : add Gaussian noise (mean=0 sigma=1)
+ -range : set the output calmin/max to full data range
+
+Matrix operations:
+ -tensor_decomp : convert a 4D (6-timepoint )tensor image into L1,2,3,FA,MD,MO,V1,2,3 (remaining image in pipeline is FA)
+
+Kernel operations (set BEFORE filtering operation if desired):
+ -kernel 3D : 3x3x3 box centered on target voxel (set as default kernel)
+ -kernel 2D : 3x3x1 box centered on target voxel
+ -kernel box    <size>     : all voxels in a cube of width <size> mm centered on target voxel
+ -kernel boxv   <size>     : all voxels in a cube of width <size> voxels centered on target voxel, CAUTION: size should be an odd number
+ -kernel boxv3  <X> <Y> <Z>: all voxels in a cuboid of dimensions X x Y x Z centered on target voxel, CAUTION: size should be an odd number
+ -kernel gauss  <sigma>    : gaussian kernel (sigma in mm, not voxels)
+ -kernel sphere <size>     : all voxels in a sphere of radius <size> mm centered on target voxel
+ -kernel file   <filename> : use external file as kernel
+
+Spatial Filtering operations: N.B. all options apart from -s use the default kernel or that _previously_ specified by -kernel
+ -dilM    : Mean Dilation of non-zero voxels
+ -dilD    : Maximum Dilation of non-zero voxels (emulating output of fslmaths 6.0.1, max not modal)
+ -dilF    : Maximum filtering of all voxels
+ -dilall  : Apply -dilM repeatedly until the entire FOV is covered
+ -ero     : Erode by zeroing non-zero voxels when zero voxels found in kernel
+ -eroF    : Minimum filtering of all voxels
+ -fmedian : Median Filtering 
+ -fmean   : Mean filtering, kernel weighted (conventionally used with gauss kernel)
+ -fmeanu  : Mean filtering, kernel weighted, un-normalized (gives edge effects)
+ -s <sigma> : create a gauss kernel of sigma mm and perform mean filtering
+ -subsamp2  : downsamples image by a factor of 2 (keeping new voxels centered on old)
+ -subsamp2offc  : downsamples image by a factor of 2 (non-centered)
+
+Dimensionality reduction operations:
```

Comment:
Dimensionalty reduction, subsamp, etc. do not exist in WASM. Only commands listed as `YES` in [this table](https://github.com/rordenlab/niimath/blob/development/src/JavaScript.md) are supported.